src/v2/components/FireList.jsx

```diff
@@ -89,6 +90,25 @@ export default function FireList(props) {
     }
   }, [nFires, selectedIndex])
 
+  const handleVoteForFire = useCallback((index, vote) => {
+    if (index > -1 && index < nFires) {
+      const fire = fires[index]
+      const {cameraID, timestamp} = fire
+      const endpoint = vote === 'undo' ? '/api/undoVoteFire' : '/api/voteFire'
+      const isRealFire = vote === 'yes'
+
+      query.post(endpoint, {cameraID, isRealFire, timestamp}).then(() => {
+        if (vote === 'undo') {
+          delete fire.voted
+        } else {
+          fire.voted = vote === 'yes'
+        }
```

Comment:
What about failure case?  Typically failures happen due to cookie errors or if UI state doesn't match server.  That's why I had put in reload() to try to resolve.

Reply:
We should handle failures of course, but I think that’s a larger (and somewhat orthogonal) change that I would like to implement in a separate PR. For now, I was leaving the error unhandled so it would be logged to the Console at least. I can add a comment to that effect—or log errors manually/explicitly. (Similarly, I still need to implement the sign-in flow started by clicking a vote button.)

As for `window.reload()`, I don’t think it’s sufficient (failures can also arise from network connectivity, DB failure, ...) and it may have less than desirable side effects—especially vis a vis UI stability (for example, reloading the page will impact scroll position and potentially introduce a new detection event—or age out others). Once we reload the page, we will lose all app state (unless we cache it in local storage) so the app and the user would be out of options to recover from the error. Additionally, the user will have no idea why the page suddenly reloaded (unless we somehow communicate the reason before or after reload). Finally—although I don’t see a problematic codepath currently—in the worst case scenario, automatically reloading to recover from errors could result in a reload loop.

“Cookie errors” smells like a bug to me. Though I can’t rule them out, I’ve never seen the sort you are suggesting in all the other apps I have worked on. It’s probably a good idea to understand why cookies aren’t synchronized correctly—I suspect that if there is a reproducible synchronization issue, it should be correctable (without user intervention or page reload).

Reply:
Case in point regarding cookie errors and page reloads: I just signed in on a separate tab using the v1 site (because I haven’t implemented the v2 sign-in workflow). Before I signed in, I couldn’t vote on the v2 tab (“Failed to load resource: the server responded with a status of 401 (Unauthorized)” as expected). After I signed in—and without reloading anything—voting just worked after I switched to the v2 tab.